src/vt-tv/api/object_communicator.h

```diff
@@ -0,0 +1,128 @@
+#if !defined INCLUDED_VT_TV_API_OBJECT_COMMUNICATOR_H
+#define INCLUDED_VT_TV_API_OBJECT_COMMUNICATOR_H
+
+#include <string>
+#include <iostream>
+#include <list>
+#include <map>
+#include <unordered_set>
+#include <vector>
+#include <utility>
+
+#include "vt-tv/api/types.h"
+#include <fmt-vt/format.h>
+
+namespace vt::tv {
+
+struct Object;
+
+/**
+ *  A class holding received and sent messages for an object.
+ */
+struct ObjectCommunicator
+{
+private:
+  ElementIDType object_id_;
+  std::map<ElementIDType, double> received_;
+  std::map<ElementIDType, double> sent_;
+
+  /**
+   * Summarize one-way communicator properties and check for errors.
+   */
+  std::vector<double> summarize_unidirectional(std::string direction) const {
+    // Initialize list of volumes
+    std::vector<double> volumes;
+
+    // Iterate over one-way communications
+    std::map<ElementIDType, double> communications;
+    if(direction == "to") {
+      communications = this->sent_;
+    } else {
+      communications = this->received_;
+    }
+
+    // for (const auto& [key, value] : communications) {
+    //   sanity check
+    //   if(key->get_id() == this->object_id_) {
+    //     throw nb::index_error(
+    //       "object " + this->object_id_ +
+    //       " cannot send communication to itself.");
+    //   }
+    //   volumes.push_back(value);
+    // }
+
+    return volumes;
+  }
+public:
+  /**
+   * \brief get id of object for this communicator
+   *
+   * \return id
+   */
+  ElementIDType get_object_id() { return object_id_; };
```

Comment:
You are using snake case here. Maybe use camel case to be consistent with other methods.